processor/metricextractprocessor/factory.go

```diff
@@ -0,0 +1,66 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricextractprocessor
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/observiq/observiq-otel-collector/internal/expr"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+)
+
+const (
+	// typeStr is the value of the "type" key in configuration.
+	typeStr = "metricextract"
+
+	// stability is the current state of the processor.
+	stability = component.StabilityLevelAlpha
+)
+
+// NewFactory creates a new factory for the processor.
+func NewFactory() component.ProcessorFactory {
+	return component.NewProcessorFactory(
+		typeStr,
+		createDefaultConfig,
+		component.WithLogsProcessor(createLogsProcessor, stability),
+	)
+}
+
+// createLogsProcessor creates a log processor.
+func createLogsProcessor(_ context.Context, params component.ProcessorCreateSettings, cfg component.ProcessorConfig, consumer consumer.Logs) (component.LogsProcessor, error) {
+	processorCfg, ok := cfg.(*Config)
+	if !ok {
+		return nil, fmt.Errorf("invalid config type: %+v", cfg)
+	}
+
+	match, err := expr.CreateBoolExpression(processorCfg.Match)
```

Comment:
@jmwilliams89 minor and non blocking question - would these make more sense in the validation of the config?

Reply:
Sure, I think we can include that.

Reply:
@jmwilliams89 now that you've put them in config validation, did you want to remove them from here?

Reply:
talked about this offline